Gorilla mux router usage fix

MatejB opened the pull request:
Hi, when I tried gom on my application that is using Gorilla Mux router it wasn't working out of the box.
I had to extract pprofstats handler so it can be manually registered to router.

Please review my pull request.

rakyll commented on docs/usage-gorilla-mux.md:
```diff
@@ -0,0 +1,32 @@
+}
+
+func attachProfiler(router *mux.Router) {
+	router.HandleFunc("/debug/pprof/", pprof.Index)
```
I was wondering whether this could be a helper function exported from gom/http/mux.

MatejB replied:
Good point, I will hack something tonight ;)

MatejB commented:
Hi, 

adding helper function was fun :)

I was not fully aware yesterday that this issue not only affects GorillaMux but any custom router instance because pprof registers it's http routes to http.DefaultServeMux in init func.

Now I added support for any custom router as long it has same signature for Handle and HandleFunc methods. Which of course GorillaMux does not so adapter was needed. 

I put everything in gom/http package because users of gom would have to import more than one package if there are using custom router instance.

I'm not pleased with signature of http.AttachProfiler function, second parameter controls registration of pprof http routes. We can't easy change this later because it could break others people apps.
If you have any suggestions pls tell.

MatejB commented:
I changed signature of http.AttachProfiler method, batter to have small code duplication (registration of /debug/pprofstats in init and AttachProfiler) than wird signature of http.AttachProfiler. 
Also moved GorillaMux adapter to separate sub package.

rakyll commented on http/http.go:
```diff
@@ -32,18 +31,44 @@ type stats struct {
+}
+
+// router
+type router interface {
```
Which other third party packages can satisfy this interface out of the box? My suggestion was to avoid adding more abstractions but to provide an http/mux package with a mux specific implementation.
```
package mux

func AttachProfiler(r *mux.Router) {}
```

There is an issue filed for martini (see https://github.com/rakyll/gom/issues/17) and I guess other frameworks can follow the same strategy to provide attacher utilities.

rakyll commented:
Given it a second thought, I decided to follow the following strategy:
- Export the Stats handler.
- Proxy pprof handler through the Stats handler, e.g. /debug/_gom/pprof/profile serves the /debug/pprof/profile handler.
- Document that the stats must be handled.

MatejB commented:
Thats sounds good.

rakyll commented:
Partially addressed the issue via b1f21705888d63b397896a18097f14f500eae30f, need to document usage.

rakyll commented:
Closed via b1f2170 and d8c1291ecccb7eea089a27db3c60206810e88e0d.
